src/pages/development/components/app-server.md

```diff
@@ -0,0 +1,196 @@
+
+### Cookies, sessions, and superglobals
+
+Using cookies, sessions, and superglobals can be a challenge due to Swoole's asynchronous, long-running server environment, which differs significantly from the traditional, stateless PHP-FPM model. Since Swoole keeps the PHP application in memory across requests, traditional PHP superglobals like `$_GET`, `$_POST`, and `$_SESSION` do not automatically reset between requests, potentially leading to data leakage or incorrect data being served to users.
+
+Additionally, implementing session management and cookie handling requires careful consideration to ensure thread safety and data isolation among concurrent requests, posing a significant challenge in preserving the state and security of a Swoole application.
+
+### Testing and debugging
+
+Traditional debugging tools and techniques designed for synchronous PHP scripts may not work as expected, as they might not properly handle the concurrency and parallel execution of tasks within Swoole. Developers must adopt or develop tools that are capable of understanding and interacting with asynchronous operations, as well as establish testing environments that can simulate the concurrent execution of multiple requests. This requires a shift in approach to both debugging and testing, with an emphasis on asynchronous logic, co-routine lifecycle management, and the potential for shared-state issues, making the process more challenging than in traditional PHP environments.
+
+## Common reasons for incompatibility
+
+1. [Not following technical guidelines](../../coding-standards/technical-guidelines.md)
+
+  - 2.9. Service classes (ones that provide behavior but not data, like EventManager) SHOULD NOT have a mutable state. To make extensions codebase compliant with technical guidelines mutable state should be removed from Service classes. Sometime it could be a property cache introduced for better performance when dealing with data-intensive operations, such as database queries, API calls, or complex calculations, that do not need to be executed repeatedly within the same request. However it brings issues with Application Server as property cache would be used in a consequent requests and it should be reseted after each request.
+
+  - 2.14. Temporal coupling MUST be avoided. Changed state at one point in time can inadvertently affect the behavior of subsequent operations, requiring a specific order of execution for the application to function correctly. This coupling makes the code harder to understand and maintain, as the correct operation of the system becomes dependent on the sequence in which state-modifying actions are performed.
+
+1. Superglobals and native PHP functions usage for header, session, cookie.
+
+  Usage of PHP Superglobals, like `$_GET`, `$_POST`, and `$_SESSION`, and native PHP functions for header, session, and cookie instead of utilizing interfaces and service contracts through dependency injection.
```

Comment:
```suggestion
  We recommend using PHP Superglobals, like `$_GET`, `$_POST`, and `$_SESSION`, and native PHP functions for header, session, and cookie, instead of utilizing interfaces and service contracts through dependency injection.
```